### PR TITLE
[config-plugins] Replace getAppThemeLightNoActionBarGroup by getAppThemeGroup

### PR DESCRIPTION
**Pull request by zoontek**
# Why

For implementing Edge to edge, I need to change the `AppTheme` parent (for `Theme.EdgeToEdge`).

Nearly all the time `assignStylesValue` is used with `getAppThemeLightNoActionBarGroup`. The issue here is that it matches both theme `name` (`AppTheme`) and `parent` (`Theme.AppCompat.Light.NoActionBar`) + `setStylesItem`, a function called exclusively by `assignStylesValue`, has [a side effect that create the theme if it does not exist](https://github.com/expo/expo/blob/main/packages/%40expo/config-plugins/src/android/Styles.ts#L91).

In short, if you change the `AppTheme` `parent`, it recreates another `AppTheme` on prebuild that inherits from `Theme.AppCompat.Light.NoActionBar`.

# How

Stop matching on both style name and parent (`getAppThemeLightNoActionBarGroup`), but only on parent (`getAppThemeGroup`).

# Test Plan

- Run a prebuild with `--clean`
- Update the `AppTheme` parent from something else (`Theme.AppCompat.DayNight.NoActionBar`, for example)
- Re-run a prebuild, this time without `--clean`
- It doesn't append another `AppTheme`

# Checklist

- [x] Documentation is up to date to reflect these changes (eg: https://docs.expo.dev and README.md).
- [x] Conforms with the [Documentation Writing Style Guide](https://github.com/expo/expo/blob/main/guides/Expo%20Documentation%20Writing%20Style%20Guide.md)
- [x] This diff will work correctly for `npx expo prebuild` & EAS Build (eg: updated a module plugin).

### CODE REVIEW

File: packages/@expo/config-plugins/CHANGELOG.md [hunk 1]
```diff
@@ -11,12 +11,13 @@
 ### 💡 Others
 
 - Upgrade `glob@7` to `glob@10`. ([#29894](https://github.com/expo/expo/pull/29894) by [@byCedric](https://github.com/byCedric))
+- Don't recreate `AppTheme` on prebuild if its parent changed. ([#30797](https://github.com/expo/expo/pull/30797) by [@zoontek](https://github.com/zoontek))
 
 ## 8.0.8 - 2024-07-11
 
 ### 🐛 Bug fixes
 
-- On `Android`, Build settings related to gradle.properties will now be **merged** (*and deduped*) if they are `Arrays` instead of overwritten. For now the only array value is `extraMavenRepos`. This allow library plugins to use `extraMavenRepos` even if they are not the last plugin called. ([#30269](https://github.com/expo/expo/pull/30269) by [@Titozzz](https://github.com/Titozzz))
```
Comment (brentvatne) [inline packages/@expo/config-plugins/CHANGELOG.md:19]: let's revert this change

Comment (zoontek) [reply packages/@expo/config-plugins/CHANGELOG.md]: Done, sorry, I had the VScode prettier plugin enabled.

File: packages/@expo/config-plugins/CHANGELOG.md [hunk 2]
```diff
@@ -11,6 +11,7 @@
 ### 💡 Others
 
 - Upgrade `glob@7` to `glob@10`. ([#29894](https://github.com/expo/expo/pull/29894) by [@byCedric](https://github.com/byCedric))
+- Don't recreate `AppTheme` on prebuild if its parent changed. ([#30797](https://github.com/expo/expo/pull/30797) by [@zoontek](https://github.com/zoontek))
```
Comment (Kudo) [inline packages/@expo/config-plugins/CHANGELOG.md:14]:
```suggestion
- Deprecated the Android style method `getAppThemeLightNoActionBarGroup` and replaced it with `getAppThemeGroup`, which does not override the parent theme if its parent theme changes. ([#30797](https://github.com/expo/expo/pull/30797) by [@zoontek](https://github.com/zoontek))
```

nit: trying to make the changelog more clear

File: packages/@expo/prebuild-config/CHANGELOG.md [hunk 3]
```diff
@@ -10,6 +10,7 @@
 
 - Fix empty splash config resulting in build error. ([#29497](https://github.com/expo/expo/pull/29497) by [@aleqsio](https://github.com/aleqsio))
 - Fix incorrect dependency imports. ([#30553](https://github.com/expo/expo/pull/30553) by [@byCedric](https://github.com/byCedric))
+- Don't recreate `AppTheme` on prebuild if its parent changed. ([#30797](https://github.com/expo/expo/pull/30797) by [@zoontek](https://github.com/zoontek))
```
Comment (Kudo) [inline packages/@expo/prebuild-config/CHANGELOG.md:13]:
```suggestion
- Replaced the config-plugins deprecated `getAppThemeLightNoActionBarGroup` method with the new `getAppThemeGroup`. ([#30797](https://github.com/expo/expo/pull/30797) by [@zoontek](https://github.com/zoontek))
```

nit: trying to make the changelog more precise

File: packages/expo-navigation-bar/CHANGELOG.md [hunk 4]
```diff
@@ -9,6 +9,7 @@
 ### 🐛 Bug fixes
 
 - Add missing `react`/`react-native` peer dependencies. ([#30573](https://github.com/expo/expo/pull/30573) by [@byCedric](https://github.com/byCedric))
+- Don't recreate `AppTheme` on prebuild if its parent changed. ([#30797](https://github.com/expo/expo/pull/30797) by [@zoontek](https://github.com/zoontek))
```
Comment (Kudo) [inline packages/expo-navigation-bar/CHANGELOG.md:12]:
```suggestion
- Replaced the config-plugins deprecated `getAppThemeLightNoActionBarGroup` method with the new `getAppThemeGroup`. ([#30797](https://github.com/expo/expo/pull/30797) by [@zoontek](https://github.com/zoontek))
```

nit: trying to make the changelog more precise

File: packages/expo-system-ui/CHANGELOG.md [hunk 5]
```diff
@@ -10,6 +10,7 @@
 
 - Add missing `react-native` peer dependencies for isolated modules. ([#30485](https://github.com/expo/expo/pull/30485) by [@byCedric](https://github.com/byCedric))
 - Add missing `react-native-web` optional peer dependency for isolated modules. ([#30689](https://github.com/expo/expo/pull/30689) by [@byCedric](https://github.com/byCedric))
+- Don't recreate `AppTheme` on prebuild if its parent changed. ([#30797](https://github.com/expo/expo/pull/30797) by [@zoontek](https://github.com/zoontek))
```
Comment (Kudo) [inline packages/expo-system-ui/CHANGELOG.md:13]:
```suggestion
- Replaced the config-plugins deprecated `getAppThemeLightNoActionBarGroup` method with the new `getAppThemeGroup`. ([#30797](https://github.com/expo/expo/pull/30797) by [@zoontek](https://github.com/zoontek))
```

nit: trying to make the changelog more precise

**Review by Kudo — COMMENTED** [review]
the code change looks promising! thanks for helping this.
just leaving some nit changelog comment.

also there's some change already included on #30835. maybe you can try rebase on main

**Comment by Kudo** [comment]
@zoontek you may need to run `yarn prepare` in the **packages/@expo/config-plugins` directory to rebuild js files. that would fix the error for sdk ci checks: https://github.com/expo/expo/actions/runs/10284137421/job/28459530021?pr=30797

**Comment by zoontek** [comment]
@Kudo Done too. BTW, why are you commiting the built files to the repository (you probably have a solid reason)? It makes contributing harder.

**Comment by Kudo** [comment]
> BTW, why are you commiting the built files to the repository (you probably have a solid reason)? It makes contributing harder.

it's something about balancing the requirement for user without typescript as well as our packages release workflow. we've discussed this internally for a long time but so far we didn't have a well and concrete solution yet. sorry for the mess.

**Review by Kudo — APPROVED** [review]
🚀🚀🚀
